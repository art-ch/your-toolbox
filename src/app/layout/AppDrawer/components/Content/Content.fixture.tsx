import { LuCalculator } from 'react-icons/lu';
import { MdMiscellaneousServices } from 'react-icons/md';

export const APP_DRAWER_CONTENT = [
  {
    icon: <LuCalculator />,
    title: 'Calculators',
    url: '/calculators',
    items: [
      {
        title: 'Ascetic Exercise Duration',
        url: '/calculators/ascetic-exercise-duration'
      }
    ]
  },
  {
    icon: <MdMiscellaneousServices />,
    title: 'Specialized Tools',
    url: '/specialized-tools',
    items: [
      {
        title: 'Rail Signalling Info',
        url: 'https://rail-signalling-info.netlify.app/'
      }
    ]
  }
];
