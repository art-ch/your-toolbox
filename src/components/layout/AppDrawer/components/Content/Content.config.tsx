import { LuCalculator } from 'react-icons/lu';
import { MdMiscellaneousServices } from 'react-icons/md';

export const APP_DRAWER_CONTENT = [
  {
    icon: <LuCalculator />,
    translationKey: 'common:calculators',
    url: '/calculators',
    items: [
      {
        title: 'Ascetic Exercise Duration',
        url: '/calculators/ascetic-exercise-duration'
      }
    ]
  },
  {
    icon: <MdMiscellaneousServices />,
    translationKey: 'common:specializedTools',
    url: '/specialized-tools',
    items: [
      {
        title: 'Rail Signalling Info',
        url: 'https://rail-signalling-info.netlify.app/'
      }
    ]
  }
];
