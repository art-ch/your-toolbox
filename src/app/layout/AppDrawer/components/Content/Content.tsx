import React from 'react';

import {
  SidebarContent,
  SidebarGroup,
  SidebarMenu,
  SidebarMenuButton,
  SidebarMenuItem,
  SidebarMenuSub,
  SidebarMenuSubButton,
  SidebarMenuSubItem,
  SidebarProvider
} from '@/components/ui/sidebar';
import { APP_DRAWER_CONTENT } from './Content.fixture';

export const Content = () => {
  return (
    <SidebarProvider className="min-h-0">
      <SidebarContent>
        <SidebarGroup>
          <SidebarMenu>
            {APP_DRAWER_CONTENT.map((item) => (
              <SidebarMenuItem key={item.title}>
                <SidebarMenuButton asChild>
                  <a href={item.url} className="font-medium">
                    {item?.icon}
                    {item.title}
                  </a>
                </SidebarMenuButton>
                {item.items?.length ? (
                  <SidebarMenuSub className="ml-0 border-l-0 px-1.5">
                    {item.items.map((item) => (
                      <SidebarMenuSubItem key={item.title}>
                        <SidebarMenuSubButton asChild>
                          <a href={item.url}>{item.title}</a>
                        </SidebarMenuSubButton>
                      </SidebarMenuSubItem>
                    ))}
                  </SidebarMenuSub>
                ) : null}
              </SidebarMenuItem>
            ))}
          </SidebarMenu>
        </SidebarGroup>
      </SidebarContent>
    </SidebarProvider>
  );
};
