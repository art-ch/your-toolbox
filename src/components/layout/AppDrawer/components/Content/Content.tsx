'use client';

import React from 'react';

import {
  SidebarContent,
  SidebarGroup,
  SidebarMenu,
  SidebarMenuButton,
  SidebarMenuItem,
  SidebarMenuSub,
  SidebarMenuSubButton,
  SidebarMenuSubItem,
  SidebarProvider
} from '@/components/ui/sidebar';
import { APP_DRAWER_CONTENT } from './Content.config';
import { useTranslation } from 'react-i18next';

export const Content = () => {
  const { t } = useTranslation();

  return (
    <SidebarProvider className="min-h-0">
      <SidebarContent>
        <SidebarGroup>
          <SidebarMenu>
            {APP_DRAWER_CONTENT.map(
              ({ translationKey, url, icon, menuOptions }) => (
                <SidebarMenuItem key={translationKey}>
                  <SidebarMenuButton asChild>
                    <a href={url} className="font-medium">
                      {icon}
                      {t(translationKey)}
                    </a>
                  </SidebarMenuButton>
                  {menuOptions?.length ? (
                    <SidebarMenuSub className="ml-0 border-l-0 px-1.5">
                      {menuOptions.map((menuOption) => {
                        const linkText =
                          // use translationKey if it exists, or title if translation is not available
                          'translationKey' in menuOption
                            ? t(menuOption.translationKey)
                            : menuOption.title;

                        return (
                          <SidebarMenuSubItem key={linkText}>
                            <SidebarMenuSubButton asChild>
                              <a href={menuOption.url}>{linkText}</a>
                            </SidebarMenuSubButton>
                          </SidebarMenuSubItem>
                        );
                      })}
                    </SidebarMenuSub>
                  ) : null}
                </SidebarMenuItem>
              )
            )}
          </SidebarMenu>
        </SidebarGroup>
      </SidebarContent>
    </SidebarProvider>
  );
};
