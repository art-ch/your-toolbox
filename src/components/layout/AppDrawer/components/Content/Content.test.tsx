import { render } from '@testing-library/react';
import { Content } from './Content';

// Mock the sidebar components
jest.mock('@/components/ui/sidebar', () => ({
  SidebarContent: ({ children }: { children: React.ReactNode }) => (
    <div data-testid="sidebar-content">{children}</div>
  ),
  SidebarGroup: ({ children }: { children: React.ReactNode }) => (
    <div data-testid="sidebar-group">{children}</div>
  ),
  SidebarMenu: ({ children }: { children: React.ReactNode }) => (
    <div data-testid="sidebar-menu">{children}</div>
  ),
  SidebarMenuItem: ({ children }: { children: React.ReactNode }) => (
    <div data-testid="sidebar-menu-item">{children}</div>
  ),
  SidebarMenuButton: ({ children }: { children: React.ReactNode }) => (
    <div data-testid="sidebar-menu-button">{children}</div>
  ),
  SidebarMenuSub: ({
    children,
    className
  }: {
    children: React.ReactNode;
    className: string;
  }) => (
    <div data-testid="sidebar-menu-sub" className={className}>
      {children}
    </div>
  ),
  SidebarMenuSubItem: ({ children }: { children: React.ReactNode }) => (
    <div data-testid="sidebar-menu-sub-item">{children}</div>
  ),
  SidebarMenuSubButton: ({ children }: { children: React.ReactNode }) => (
    <div data-testid="sidebar-menu-sub-button">{children}</div>
  ),
  SidebarProvider: ({
    children,
    className
  }: {
    children: React.ReactNode;
    className: string;
  }) => (
    <div data-testid="sidebar-provider" className={className}>
      {children}
    </div>
  )
}));

// Mock the Content.config with updated structure
jest.mock('./Content.config', () => ({
  APP_DRAWER_CONTENT: [
    {
      translationKey: 'common:calculators',
      url: '/calculators',
      icon: null,
      menuOptions: []
    },
    {
      translationKey: 'common:specializedTools',
      url: '/specialized-tools',
      icon: null,
      menuOptions: [
        { translationKey: 'tool1:title', url: '/tool1' },
        { title: 'Tool 2', url: '/tool2' }
      ]
    }
  ]
}));

// Mock react-i18next
jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: (key: string) => key
  })
}));

describe('Content component', () => {
  it('should render correctly', () => {
    const { container } = render(<Content />);

    expect(container).toMatchSnapshot();
  });
});
