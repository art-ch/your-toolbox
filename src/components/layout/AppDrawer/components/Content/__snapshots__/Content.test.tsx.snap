// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content component should render correctly 1`] = `
<div>
  <div
    class="min-h-0"
    data-testid="sidebar-provider"
  >
    <div
      data-testid="sidebar-content"
    >
      <div
        data-testid="sidebar-group"
      >
        <div
          data-testid="sidebar-menu"
        >
          <div
            data-testid="sidebar-menu-item"
          >
            <div
              data-testid="sidebar-menu-button"
            >
              <a
                class="font-medium"
                href="/calculators"
              >
                common:calculators
              </a>
            </div>
          </div>
          <div
            data-testid="sidebar-menu-item"
          >
            <div
              data-testid="sidebar-menu-button"
            >
              <a
                class="font-medium"
                href="/specialized-tools"
              >
                common:specializedTools
              </a>
            </div>
            <div
              class="ml-0 border-l-0 px-1.5"
              data-testid="sidebar-menu-sub"
            >
              <div
                data-testid="sidebar-menu-sub-item"
              >
                <div
                  data-testid="sidebar-menu-sub-button"
                >
                  <a
                    href="/tool1"
                  >
                    tool1:title
                  </a>
                </div>
              </div>
              <div
                data-testid="sidebar-menu-sub-item"
              >
                <div
                  data-testid="sidebar-menu-sub-button"
                >
                  <a
                    href="/tool2"
                  >
                    Tool 2
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
